common/util/autoware_utils/include/autoware_utils/point_types/float_comparisons.hpp

```diff
@@ -0,0 +1,148 @@
+// Copyright 2020 Mapless AI, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to
+// deal in the Software without restriction, including without limitation the
+// rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+// sell copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+
+#ifndef HELPER_FUNCTIONS__FLOAT_COMPARISONS_HPP_
+#define HELPER_FUNCTIONS__FLOAT_COMPARISONS_HPP_
+
+#include <algorithm>
+#include <cmath>
+#include <limits>
+
+namespace autoware
+{
+namespace common
+{
+namespace helper_functions
+{
+namespace comparisons
```

Comment:
We should consider the rule of C++ namespace. I think it's good to make it the same as the package name.

Reply:
This file is copied from Autoware.Auto so I don't want to change the code if possible.

https://gitlab.com/autowarefoundation/autoware.auto/AutowareAuto/blob/master/src/common/autoware_auto_common/include/common/types.hpp

Reply:
Then, unfortunately, I think you can't merge this to `autoware_utils` because it's not consistent.

Reply:
It was resolved by https://github.com/tier4/AutowareArchitectureProposal.iv/pull/784